synthesis/population/spatial/primary/candidates.py

```diff
@@ -23,6 +23,7 @@ def sample_destination_municipalities(context, arguments):
     df_od = df_od[df_od["origin_id"] == origin_id].copy()
 
     # Sample destinations
+    df_od["weight"] = df_od["weight"].map(lambda x: float(x))
```

Comment:
Is this needed in this PR or this is rather a fix for something else?

Reply:
This was indeed a fix we had to add on our fork, so it is not related to this PR. I removed it. Seems to run fine still.